src/sentry/migrations/0232_grouphistory.py

```diff
@@ -0,0 +1,77 @@
+# Generated by Django 2.2.24 on 2021-09-20 20:38
+
+import django.db.models.deletion
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0231_alert_rule_comparison_delta"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="GroupHistory",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                (
+                    "status",
+                    sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
+                        db_index=True, default=0
+                    ),
+                ),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                (
+                    "group",
```

Comment:
What should happen to `GroupHistory` when groups are merged? With this cascading constraint history records will be deleted unless merge code is also updated.

Reply:
I think I'll probably filter out merged history at the very least, much like we do for merged activity. Still considering whether I should delete it altogether (which we don't seem to do for activity, just filter out 'duplicate' (based on type, user, and ident in `get_activities_for_group`)